### DISCUSSION TITLE
tilemaker   versus   protomaps.js

### DISCUSSION THREAD

**Discussion by stephan75** [systemed/tilemaker#334] [General]
Hello,
in my github notifications watchlist I have several projects related to OSM, where tilemaker is one of them.

But I have also https://github.com/protomaps/protomaps.js there ...

Though I have not enough skills and hardware to run tilemaker or protomaps.js on my own, I ask myself: 

_What is the difference between tilemaker and protomaps.js?_

So what features do they have in common, and where do they differ?

Of course I already read documentation and blogs for them both, but I am not able to put any answer in own words.

Maybe both projects can benefit from each other?

Or why are there some issues in one of them that seem to not occure in the other one?

**Comment by systemed** [comment]
Broadly speaking:

- tilemaker takes raw OSM data (.osm.pbf) and creates vector tiles (an .mbtiles container of .mvt files)
- protomaps.js renders vector tiles for on-screen display (much like MapLibre or Mapbox GL)
- [PMTiles](https://github.com/protomaps/PMTiles), another part of the Protomaps project, reserialises .mbtiles into an efficient format suitable for commodity hosting

So they actually fit together quite well: you can create an .mbtiles with tilemaker, convert it to .pmtiles, and serve it for display using protomaps.js.

(@bdon can correct me if I've misunderstood anything!)

**Reply by bdon** [reply]
Sounds right to me! [protomaps.js](https://github.com/protomaps/protomaps.js) is just another javascript rendering library, though one that is less powerful than the GL ones for simple use cases. You can download basemap PMTiles directly from me at [protomaps.com/bundles](https://protomaps.com/bundles) but for those that want to DIY the basemap tiles with an open source tool I recommend what Richard said: Create MBTiles with FileMaker + convert to PMTiles.
